src/main/app/src/components/Purchasing.js

```diff
@@ -41,25 +54,163 @@ const orderRows = [
   { id: 4, vendorName: 'Vendor1', items: '????', timestamp: "01/31/2021", cost: "$100" }
 ];
 
-const LoadedView = () => {
+function TextMaskCustom(props) {
+  const { inputRef, ...other } = props;
+
+  return (
+    <MaskedInput
+      {...other}
+      ref={(ref) => {
+        inputRef(ref ? ref.inputElement : null);
+      }}
+      mask={['(', /[1-9]/, /\d/, /\d/, ')', ' ', /\d/, /\d/, /\d/, '-', /\d/, /\d/, /\d/, /\d/]}
+      placeholderChar={'\u2000'}
+      showMask
+    />
+  );
+}
+
+TextMaskCustom.propTypes = {
+  inputRef: PropTypes.func.isRequired,
+};
+
+const PhoneNumberField = () => {
+  const [values, setValues] = React.useState({
+    textmask: '(  )    -    ',
+    numberformat: '1320',
+  });
+
+  const handleChange = (event) => {
+    setValues({
+      ...values,
+      [event.target.name]: event.target.value,
+    });
+  };
+  return (
+    <FormControl>
+        <InputLabel>Phone Number</InputLabel>
+        <Input
+          value={values.textmask}
+          onChange={handleChange}
+          name="textmask"
+          inputComponent={TextMaskCustom}
+        />
+      </FormControl>
+  );
+};
+
+const ProvinceDrawer = () =>{
+  const [province, setProvince] = React.useState('');
+
+  const handleChange = (event) => {
+    setProvince(event.target.value);
+  };
+  return(
+    <div  style={{ width: "250px" }}>
+      <FormControl>
+        <Select
+          value={province}
+          onChange={handleChange}
+        >
+          <MenuItem value={"QC"}>QC</MenuItem>
```

Comment:
Can we add all provinces? 

Something like this would be easiest:
```javascript
<Select
  value={province}
  onChange={handleChange}
>
  {
    ['ON', 'QC', 'AB',
     'BC', 'NB', 'MB'
      /*... the rest of the provinces*/
    ].map(province =>
      <MenuItem value={province}>
        {province}
      </MenuItem>
    )
  }
</Select>
```

Reply:
yeah the two provinces were just placeholders, will include all provinces